test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env tarantool
```

Comment:
The test stucks for me locally:

```sh
$ LUA_PATH=test/app-tap/lua/?.lua ./src/tarantool test/app-tap/gh-2717-no-quit-sigint.test.lua
TAP version 13
1..3
Tarantool 2.10.0-beta1-357-gd2baedf0e
type 'help' for interactive help
<..here it stucks..>
<..but if I'll hit Enter, it'll continue..>
ok - SIGINT doesn't kill tarantool in interactive mode 
ok - Ctrl+C discards the input and calls the new prompt
ok - SIGINT terminates tarantool in daemon mode
```

BTW, the terminal is left in an inconsistent state after this (requires `reset`).

The same in test-run, the test stucks:

```sh
$ ./test/test-run.py 2717
<...>
No output during 10 seconds. Will abort after 120 seconds without output. List of workers not reporting the status:
- 001_app-tap [app-tap/gh-2717-no-quit-sigint.test.lua, None] at var/001_app-tap/gh-2717-no-quit-sigint.result:0
<...>
```

Reply:
It is on `output = output .. ph:read()` after a loop, on line 39.

Reply:
I don't like that a test may break terminal options and that it depends on a real tty on stdin. I would replace `'r'` modifier with `'rw'` to give another stdin for a child process. The `-i` flag is ignored in the case (#5064), but there is a workaround:

```lua
local ph = popen.shell('ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH, 'rw')
```

Reply:
NB: Using an error injection means that the test will work only for debugging tarantool build, so it should be reflected in `release_disabled` field in `suite.ini` file.

Reply:
Done.

Reply:
Seems resolved (according to the code). I'll reopen the discussion if I'll see a problem of this kind.